### PR TITLE
GH-126 Central Dependency Management using toml file.

### PR DESCRIPTION
**Pull request by vLuckyyy**
# Using `libs.versions.toml` for Central Dependency Management

The purpose of this pull request is to establish a more organized dependency management structure in our Gradle project by introducing the use of `libs.versions.toml`

## Details

We have created the `libs.versions.toml` file in the `gradle/` directory, which is the main project directory. This file helps us define all dependency versions in one place, significantly reducing redundancy and simplifying library updates.

### For library:
obraz

### For gradle plugin:
Bez tytułu

## Benefits

- **Central Dependency Management**: All library versions are now defined in a single `libs.versions.toml` file. This streamlines version management and helps keep the project up to date.

- **Redundancy Elimination**: Previously, the same versions and dependencies had to be defined in multiple places, leading to redundancy and making management more challenging. Now, we all rely on a single source of truth.

## Conclusions

These changes simplify dependency and configuration management in our Gradle project. Now, when we want to update a library version, we only need to change it in one place, and this change automatically propagates to all modules using it. This is an official and recommended Gradle feature, as per the [Gradle documentation](https://docs.gradle.org/current/userguide/platforms.html#sub:conventional-dependencies-toml). This model is much more sustainable and easier to manage in the long run. 🚀💡

### CODE REVIEW

**Review by Jakubk15 — APPROVED** [review]

File: eternalcombat-api/build.gradle.kts [hunk 1]
```diff
@@ -89,21 +61,16 @@ publishing {
             from(project.components["java"])
         }
     }
-}
 
-publishing {
```
Comment (Rollczi) [inline eternalcombat-api/build.gradle.kts:94]: czemu publish zostało usunięte?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+jUnitJupiterApi = { group = "org.junit.jupiter", name = "junit-jupiter-api", version.ref = "junitJupiterApi" }
+jUnitJupiterParams = { group = "org.junit.jupiter", name = "junit-jupiter-params", version.ref = "junitJupiterParams" }
+jUnitJupiterEngine = { group = "org.junit.jupiter", name = "junit-jupiter-engine", version.ref = "junitJupiterEngine" }
+
+[plugins]
+
+pluginyml = { id = "net.minecrell.plugin-yml.bukkit", version.ref = "pluginyml" }
+shadow = { id = "com.github.johnrengelman.shadow", version.ref = "shadowjar" }
+runpaper = { id = "xyz.jpenilla.run-paper", version.ref = "runpaper" }
```
Comment (Rollczi) [inline gradle/libs.versions.toml:58]: teraz żeby prześledzić jakiekolwiek zależności per moduł to trzeba zaglądać ciągle do tego pliku. jedyne co by mogło mieć sens to wersjonowanie ale takie coś to przesada kompletnie

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
+}
+
+tasks.compileJava {
+    options.compilerArgs = listOf("-Xlint:deprecation", "-parameters")
+    options.encoding = "UTF-8"
+}
```
Comment (Rollczi) [inline build.gradle.kts:40]: nie musisz tego centralizować, po prostu dodaj konwencje w `buildSrc`

**Review by Rollczi — CHANGES_REQUESTED** [review]
centralizacja jak w starym dobry komunizmie

**Review by imDMK — APPROVED** [review]

**Review by vLuckyyy — COMMENTED** [review]
I will merge it, I understand that not everyone likes this convention, we will introduce any corrections in separate pull requests, currently the gradle setup on the master is not done properly, **so it prevents any work on the plugin.**

**Pull request merged**